src/app/components/OrderInput.tsx

```diff
@@ -0,0 +1,435 @@
+        <span className="label-text-alt text-error">
+          {validationResult.valid ? "" : validationResult.message}
+        </span>
+      </label>
+
+      <div className="btn-group w-full">
+        <button
+          className="btn btn-sm"
+          onClick={() => dispatch(setSizePercent(25))}
+        >
+          25%
+        </button>
+        <button
+          className="btn btn-sm"
+          onClick={() => dispatch(setSizePercent(50))}
+        >
+          50%
+        </button>
+        <button
+          className="btn btn-sm"
+          onClick={() => dispatch(setSizePercent(75))}
+        >
+          75%
+        </button>
+        <button
+          className="btn btn-sm"
+          onClick={() => dispatch(setSizePercent(100))}
+        >
+          100%
+        </button>
+        <input
+          type="number"
+          className="input input-sm bg-neutral"
+          onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
+            const size = Number(event.target.value);
+            dispatch(setSizePercent(size));
+          }}
+        ></input>
+      </div>
+    </div>
+  );
+}
+
+// TODO: test if floating point numbers are handled correctly
+function slippagePercentage(slippage: number): string {
+  return displayNumber(slippage * 100, 0);
+}
+
+function slippageFromPercentage(percentage: string): number {
+  // TODO: decimal numbers with dots (1.3) don't work (but 1,3 does)
+  return Number(percentage) / 100;
+}
+
+function MarketOrderInput() {
+  const defaultValue = useAppSelector((state) => state.orderInput.slippage);
+  const validationResult = useAppSelector(validateSlippageInput);
+  const dispatch = useAppDispatch();
+  return (
+    <>
+      <div className="form-control">
+        <label className="label">
+          <span className="label-text">Slippage</span>
+        </label>
+        <div className="relative">
+          <input
+            type="number"
+            className={
+              "input input-bordered w-full" +
+              (validationResult.valid ? "" : " input-error")
+            }
+            defaultValue={slippagePercentage(defaultValue)}
+            onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
+              const slippage = slippageFromPercentage(event.target.value);
+              dispatch(orderInputSlice.actions.setSlippage(slippage));
+            }}
+          ></input>
+          <div className="absolute right-2 top-1/2 -translate-y-1/2 flex items-center space-x-2">
+            <span>%</span>
+          </div>
+        </div>
+        <label className="label">
+          <span className="label-text-alt text-error">
+            {validationResult.valid ? "" : validationResult.message}
+          </span>
+        </label>
+      </div>
+
+      <PositionSizeInput />
+    </>
+  );
+}
+
+function LimitOrderInput() {
+  const defaultValue = useAppSelector((state) => state.orderInput.price);
+  const priceToken = useAppSelector(getUnselectedToken);
```

Comment:
Minor thing, but until we implement inverted pricing this should just be token1?

Reply:
I don't think it is. The price is how much you are willing to give (or take) in exchange for 1 selected token, so it should always be in terms of unselected token... or am I missing something?